src/javascript/app_2/pages/trading/components/elements/tabs.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+
+class Tabs extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = { active: '1' };
+    }
+    render() {
+        const className = 'tab-container';
+        return (
+            <div className={`${className} ${this.props.alignment}`}>
+                <TabsWrapper
+                  active={this.state.active}
+                  onChange={active => this.setState({active})}
+                >
+                {Object.keys(this.props.list).map(key => (
+                    <React.Fragment key={key}>
+                        {!!this.props.list[key].icon &&
+                            <i className={this.props.list[key].icon}></i>
+                        }
+                        <span>{this.props.list[key].header}</span>
+                    </React.Fragment>
+                ))}
+                </TabsWrapper>
+                <p className='tab-content'>{this.props.list[this.state.active].content}</p>
+            </div>
+        );
+    }
+};
+
+class TabsWrapper extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            sizes: {},
+        };
+        this.els = {};
+    }
+
+    componentDidMount() {
+        this.getSizes();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.children !== this.props.children && prevProps.active !== this.props.active) {
+            this.getSizes();
```

Comment:
`this.getSizes()` updates the state which in turn causes the component to rerender.  
With the `if (prevProps.children !== this.props.children && prevProps.active !== this.props.active)` it's not a dealbreaker since it will only render twice.

I don't have a better solution!
Could you add a brief comment `componentDidUpdate()`?
Maybe later we can figure out a better way to implement it.

Reply:
Sure, I'll add the comment.